src/components/ItemsList.js

```diff
@@ -23,13 +32,33 @@ const ItemsList = () => {
     { idField: 'documentId' },
   );
 
-
   const markItemAsPurchased = (index) => {
     const { items, documentId } = shoppingList[0];
-
-    items[index].lastPurchasedOn = items[index].lastPurchasedOn
-      ? null
-      : Date.now();
+    const { lastPurchasedOn } = items[index];
+    const shoppingItemObject = items[index];
+    if (lastPurchasedOn === null) {
+      shoppingItemObject.lastPurchasedOn = Date.now();
+      shoppingItemObject.numberOfPurchases++;
+      shoppingItemObject.daysLeftForNextPurchase = calculateEstimate(
+        undefined,
+        shoppingItemObject.daysLeftForNextPurchase,
+        shoppingItemObject.numberOfPurchases,
+      );
+    } else {
+      if (wasItemPurchasedWithinLastOneDay(lastPurchasedOn)) {
```

Comment:
Very tiny suggestion! But I think it's possible for line 47 to state } else if (....) { just removing the need for the if statement being on a separate line.

Reply:
thanks Becky, we structured it in such a way that the if else will run only when the condition for else is true. and also there's another if statement that will run if the other if evaluates to true. so for readability we separated them that way